lib/Data/Validate/Sanctions/Fetcher.pm

```diff
@@ -4,32 +4,83 @@ use strict;
 use warnings;
 
 use DateTime::Format::Strptime;
+use Date::Utility;
 use IO::Uncompress::Unzip qw(unzip $UnzipError);
-use List::Util qw/uniq/;
+use List::Util qw(uniq any);
 use Mojo::UserAgent;
 use Text::CSV;
-use Try::Tiny;
+use Syntax::Keyword::Try;
 use XML::Fast;
 
 our $VERSION = '0.10';
 
-my $config = {
-    'OFAC-SDN' => {
-        description => 'TREASURY.GOV: Specially Designated Nationals List with a.k.a included',
-        url    => 'https://www.treasury.gov/ofac/downloads/sdn_xml.zip',    #let's be polite and use zippped version of this 7mb+ file
-        parser => \&_ofac_xml_zip,
-    },
-    'OFAC-Consolidated' => {
-        description => 'TREASURY.GOV: Consolidated Sanctions List Data Files',
-        url         => 'https://www.treasury.gov/ofac/downloads/consolidated/consolidated.xml',
-        parser      => \&_ofac_xml,
-    },
-    'HMT-Sanctions' => {
-        description => 'GOV.UK: Financial sanctions: consolidated list of targets',
-        url         => 'https://ofsistorage.blob.core.windows.net/publishlive/ConList.csv',
-        parser      => \&_hmt_csv,
-    },
-};
+=head2 config
+
+Creastes a hash-ref of sanction source configuration, including their url, description and parser callback.
+It accepts the following list of named args:
+
+=over 4
+
+=item B<-eu_token>: required if B<eu_url> is empty
+
+The token required for accessing EU sanctions (usually added as an arg to URL).
+
+=item <eu_url>: required if B<eu_token> is empty
+
+EU Sanctions full url, token included.
+
+=item B<ofac_sdn_url>: optional
+
+OFAC-SDN download url.
+
+=item B<ofac_consolidated_url>: optional
+
+OFAC Consilidated download url.
+
+=item B<hmt_url>: optional
+
+MHT Sanctions download url.
+
+=back
+
+=cut
+
+sub config {
+    my %args = @_;
+
+    my $eu_token = $args{eu_token} // $ENV{EU_SANCTIONS_TOKEN};
+    my $eu_url   = $args{eu_url}   // $ENV{EU_SANCTIONS_URL};
+
+    warn 'EU Sanctions will fail whithout eu_token or eu_url' unless $eu_token or $eu_url;
```

Comment:
Should we  check here only `$eu_token` because few lines below we'll assign default value for `$eu_url`. And should we die here instead of warning?

Reply:
I don't think that it can be done by checking `eu_token` alone. the logic here is like this:
- if there's an `eu_token` build the default url using it
- otherwise use `eu_url` (specially useful for unit tests)
I think `warn` is better here; otherwise all sources will break because of missing EU sanctions token.